### DISCUSSION TITLE
About downloading old 「Truffle unbox react」

### DISCUSSION THREAD

**Discussion by okahijiki** [ConsenSys-archive/truffle#5552] [General Q&A] [answered]
I am currently working on a tutorial on "Hands-On Smart Contract Development With Solidity and Ethereum" (O'Reilly/Japanese edition).

In this book, the older Truffle unbox react is used instead of the latest version.

Please let me know how to download the older version.

Please refer to the following image for the older version.

[https://github.com/okahijiki/truffleFaqRef/blob/main/P94.JPG](url)

**Comment by okahijiki** [comment]
When you click on the URL for reference, an error is displayed.
Please copy the URL and paste it into the address bar.

**Comment by cliffoo** [comment, marked as answer]
Hey @okahijiki . So the process is a little more involved than the one-liner `truffle unbox react`. Here's what you can do:

1. Git clone [react-box](https://github.com/truffle-box/react-box)
2. Hard reset the cloned repo via `git reset --hard <commit-you-want>`
   * To find the commit you want, you can use just browse on github or `git log` or whatever you prefer :)
3. Then `truffle unbox <path-to-cloned-react-box-repo>`

And that should be it! Let me know if this is helpful. Cheers 🥂

\* Edit: Older iterations of react-box are rather out of date, things may not work properly (e.g. connecting to MetaMask). Unbox the latest react-box (`truffle unbox react`) if possible.

**Reply by okahijiki** [reply]
Hi! @cliffoo

Thank you very much for your answer.
I use DeepL (automatic translation) to write and read.
If I have misunderstood you or if it sounds strange, I am sorry.
You are recommending the react box, perhaps.
If so, I would like to use it when I have a chance.

On the other hand, I would like to do a tutorial following this book.
As you can see in the attached image, the truffle folder, 
for example, does not exist 「truffle」folder in the version covered in this book...

Is it possible to install a version of unbox with a file structure like the attached image?

For example

truffle unbox react@xx.xx.xx (where xx is the version number)

**Reply by cliffoo** [reply]
I can understand you just fine! Give this a try:

1. `git clone https://github.com/truffle-box/react-box.git`
2. `( cd react-box; git reset --hard c33a70f57559c0040f7109f9db498a0f83187776 )`
3. `mkdir old-react-box && cd old-react-box`
4. `truffle unbox ../react-box`

Boxes have no concept of version at the moment.

**Reply by okahijiki** [reply]
It was cleared.thank you〜

**Reply by okahijiki** [reply]
Thank you for your help on that occasion.

In this discussion, I have downloaded the tuffle unbox react folder by following the procedure you gave me.

However, after downloading, I wrote some code and tested it, but no matter how many times I try, I keep getting an error.

So, I rechecked this discussion and noticed that after step 4, the following error is displayed.

I’m sorry for the trouble. Thank you so much.

<img width="1368" alt="discussion" src="https://user-images.githubusercontent.com/22391336/193214367-e85b81cd-ddf5-47d7-9fb2-efec8bee200a.png">

**Reply by cliffoo** [reply]
Oh that's weird. Well I guess there are two things you can try:
1. I noticed that you're using an outdated version of Truffle. You can `npm uninstall -g truffle && npm install -g truffle` to get the latest release. We release every week with fixes and improvements. Then retry the steps 1 - 4 from above and see if it works.
2. As a last resort, you can avoid the `unbox` command altogether and still get what you want. So do steps 1 - 2 from above, then
    - `cd react-box`
    - `rm -rf .gitignore box-img-lg.png box-img-sm.png .github`
    - `( cd client; npm ci )`
    Then you'll have the result of running `truffle unbox react` without actually running the command!

Hopefully this helps. Let me know how it goes :)

**Reply by okahijiki** [reply]
The latter method(2） has been cleared.Thank you !
